File: C++ solutions/2470. Number of Subarrays With LCM Equal to K.cpp
class Solution {
public:
    int GCD(int A, int B) {
        if(B==0) return A;
        return GCD(B, A%B);
    }
    long long LCM(int A, int B) {
        long long ans=(A*B);
        long long res=GCD(A,B);
        return ans/res;
    }
    int subarrayLCM(vector<int>& nums, int k) {
        int ans=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==k) ans++;
            long long lcm=nums[i];
            for(int j=i+1;j<nums.size();j++){
                long long res=LCM(lcm,nums[j]);
                if(res==k) ans++;
                lcm=res;
                if(res>k) break;
            }
        }
        return ans;
    }
};

File: JavaScript solutions/2470. Number of Subarrays With LCM Equal to K.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarrayLCM = function (nums, k) {
  const GCD = (A, B) => {
    if (B == 0) return A;
    return GCD(B, A % B);
  };

  const LCM = (A, B) => {
    return (A * B) / GCD(A, B);
  };

  let ans = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === k) ans++;
    let lcm = nums[i];
    for (let j = i + 1; j < nums.length; j++) {
      let res = LCM(lcm, nums[j]);
      if (res == k) ans++;
      lcm = res;
      if (res > k) break;
    }
  }
  return ans;
};
